package/com.unity.formats.usd/Tests/Editor/EditorImportAnalyticsTests.cs

```diff
@@ -0,0 +1,212 @@
+# if UNITY_2023_3_OR_NEWER
+
+using System.Collections;
+using NUnit.Framework;
+using USD.NET;
+using UnityEngine.TestTools;
+
+namespace Unity.Formats.USD.Tests
+{
+    public class EditorImportAnalyticsTests : EditorAnalyticsBaseFixture
+    {
+        static ImportMethods[] importMethod = new ImportMethods[] { ImportMethods.AsGameObject, ImportMethods.AsPrefab, ImportMethods.AsTimelineRecording };
+
+        Scene m_scene;
+
+        [UnityTest]
+        [Retry(3)]
+        // sending valid import analytics returns ok
+        // Analytics have correct file ending for valid files imported
+        // Analytics for valid import contains non-zero timestamp
+        public IEnumerator OnValidImportExtensions_AnalyticsAreSent([ValueSource("importMethod")] ImportMethods importMethod, [ValueSource("usdExtensions")] string extension)
+        {
+            m_scene = TestUtility.CreateTestUsdScene(ArtifactsDirectoryFullPath, "testFile" + extension);
+            ImportAs(m_scene, importMethod);
+
+            AnalyticsEvent expectedEvent = null;
+            yield return (WaitForUsdAnalytics<UsdAnalyticsEventImport>(UsdAnalyticsTypes.Import, (AnalyticsEvent waitedEvent) => {
+                expectedEvent = waitedEvent;
+            }));
+
+            Assert.IsNotNull(expectedEvent);
+
+            var expectedUsdEventMsg = ((UsdAnalyticsEventImport)expectedEvent.usd).msg;
+            Assert.IsTrue(expectedUsdEventMsg.Succeeded, "Expected True for import successful");
+            Assert.AreEqual(extension, expectedUsdEventMsg.FileExtension, "Expected file extension was incorrect");
+            Assert.Greater(expectedUsdEventMsg.TimeTakenMs, 0, "Time Taken MS was reported to be 0 or less");
+        }
+
+        [UnityTest]
+        [Retry(3)]
+        // Analytics for a mesh 'contains mesh'
+        public IEnumerator OnValidImportMeshedScene_AnalyticsAreSent([ValueSource("importMethod")] ImportMethods importMethod)
+        {
+            m_scene = OpenUSDGUIDAssetScene(TestDataGuids.Material.SimpleMaterialUsd, out _);
+            ImportAs(m_scene, importMethod);
+
+            AnalyticsEvent expectedEvent = null;
+            yield return (WaitForUsdAnalytics<UsdAnalyticsEventImport>(UsdAnalyticsTypes.Import, (AnalyticsEvent waitedEvent) => {
+                expectedEvent = waitedEvent;
+            }));
+
+            Assert.IsNotNull(expectedEvent);
+
+            var expectedUsdEventMsg = ((UsdAnalyticsEventImport)expectedEvent.usd).msg;
```

Comment:
Nitpick: This isn't the 'expected', it's the 'actual', right?

Reply:
good point - will change the names